card/src/main/java/com/adyen/checkout/card/CardComponent.kt

```diff
@@ -101,6 +114,55 @@ class CardComponent private constructor(
         )
     }
 
+    private fun fetchCardType(cardNumber: String) {
+        viewModelScope.launch {
+            val deferredEncryption = viewModelScope.async(Dispatchers.Default) {
+                CardEncrypter.encryptBin(cardNumber, publicKey)
+            }
+            try {
+                val encryptedBin = deferredEncryption.await()
+                val request = BinLookupRequest(encryptedBin, UUID.randomUUID().toString(), getCardTypes())
+                val response = BinLookupConnection(request, configuration.environment, configuration.clientKey).suspendedCall()
+                cardTypeReceived(response)
+            } catch (e: EncryptionException) {
+                Logger.e(TAG, "Failed to encrypt BIN", e)
+                return@launch
+            } catch (e: IOException) {
+                Logger.e(TAG, "Failed to call binLookup API.", e)
+                return@launch
+            }
+        }
+    }
+
+    private fun cardTypeReceived(binLookupResponse: BinLookupResponse) {
+        Logger.d(TAG, "cardBrandReceived")
+        val brands = binLookupResponse.brands
+        when {
+            brands == null || brands.isEmpty() -> {
+                Logger.d(TAG, "Card brand not found.")
+                // TODO: 19/01/2021 Keep regexes prediction and don't apply business rules
+            }
+            brands.size > 1 -> {
+                Logger.d(TAG, "Multiple brands found.")
+                // TODO: 19/01/2021 use first brand
+            }
+            else -> {
+                Logger.d(TAG, "Card brand: ${brands[0].brand}")
+                val cardType = CardType.getByBrandName(brands[0].brand ?: "")
```

Comment:
Minor: you can also use `brands[0].brand.orEmpty()` or `brands.first().brand.orEmpty()`